src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt

```diff
@@ -3,18 +3,24 @@ package com.sourcegraph.cody.auth
 import com.intellij.openapi.project.Project
 import com.sourcegraph.cody.CodyToolWindowContent
 import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.config.CodyProjectActiveAccountHolder
 import com.sourcegraph.cody.config.getFirstAccountOrNull
 import com.sourcegraph.cody.initialization.Activity
 
 class SelectOneOfTheAccountsAsActive : Activity {
 
   override fun runActivity(project: Project) {
-    if (CodyAuthenticationManager.getInstance(project).hasNoActiveAccount()) {
-      val newActiveAccount =
-          CodyAuthenticationManager.getInstance(project).getAccounts().getFirstAccountOrNull()
-      CodyAuthenticationManager.getInstance(project).setActiveAccount(newActiveAccount)
+    // It is possible that CodyProjectActiveAccountHolder loads the account from the state omitting
```

Comment:
> It is possible that
Can you elaborate on that?

Reply:
```
@State(
    name = "CodyActiveAccount",
    storages = [Storage(StoragePathMacros.WORKSPACE_FILE)],
    reportStatistic = false)
@Service(Service.Level.PROJECT)
class CodyProjectActiveAccountHolder : PersistentActiveAccountHolder<CodyAccount>() {

  override fun accountManager() = service<CodyAccountManager>()

  override fun notifyActiveAccountMissing() {}

  companion object {
    @JvmStatic
    fun getInstance(project: Project): CodyProjectActiveAccountHolder {
      return project.service<CodyProjectActiveAccountHolder>()
    }
  }
}
```

Reply:
The class persists the account when we close the IDE. And sets the account back on restart skipping CodyAuthenticationManager.setActiveAccount.

Reply:
Summary

- The present solution works. 
- We cannot call setActiveAccount from loadState b/c it results with infinite recursive call (stackoverflow).
- IMO we have a guarantee that loadState happens before setActiveAccount as loadStates happens in CodyProjectActiveAccountHolder constructor (indirectly)

so the present solution is good enough, but

We should revisit this design. It seems that CodyProjectActiveAccountHolder and CodyAuthenticationManager could be merged into a single class.

Reply:
https://linear.app/sourcegraph/issue/CODY-2666/merge-codyprojectactiveaccountholder-and-codyauthenticationmanager